tests/CommentRemovalTest.hack

```diff
@@ -224,9 +224,11 @@ setTimeout(function(){
 		$clean_html3 = $purifier->purify($dirty_html3);
 		$clean_html4 = $purifier->purify($dirty_html4);
 
-		expect($clean_html1)->toEqual('--&gt;-&gt;--&gt;');
-		expect($clean_html2)->toEqual('-&gt;--&gt;');
-		expect($clean_html3)->toEqual('-&gt;--&gt;-&gt;--&gt;--&gt;-&gt;');
-		expect($clean_html4)->toEqual('-&gt;--&gt;-&gt;--&gt;--&gt;-&gt;-&gt;--&gt;--&gt;-&gt;--&gt;--&gt;-&gt;-&gt;');
+		expect($clean_html1)->toEqual("--&gt;-&gt;-&lt;!--&amp;--&gt;-&amp;gt;\n");
```

Comment:
This is the new output of the comment removal. It still works in the sense that a user trying to break the sanitization via comments and iframes won't get a usable string, but it still isn't pretty. Are there any issues with this/should we look into handling this differently?

Reply:
I think has something to do with the iframe not being properly closed... I'm gonna do some more poking around here, but fwiw when the dirty HTML is `<!--<!-<!-->-->-><iframe width=100 srcdoc="<script>alert(document.domain)</script>"/>-<!-->-->->`, the cleaned output in this test case would be `--&gt;-&gt;--&gt;`.

Reply:
Okay so further proding shows that after tokenizeHTML (which is where I added the logic), the list of tokens is:
```
vec[3]
0:HTMLPurifier\Token\HTMLPurifier_Token_Text
    data: -->->
1:HTMLPurifier\Token\HTMLPurifier_Token_Start
    iframe
2:HTMLPurifier\Token\HTMLPurifier_Token_Text
    data: -<!--&-->-&gt;
```
So it looks like the third token is based off of the substring `"-<!-->-->->"`, and the first `>` in that substring is treated as text in a comment, meaning that it gets converted to `&` per the if block at line 31 in tokenizeHTML.

If I combine these (and take out the iframe because we don't want to allow that per the policy), the remaining text tokens would be `-->->-<!--&-->-&gt;`. But since this is text, we end up escaping _again_, meaning that _all_ `&` are converted to `&amp;`, `<` to `&lt;`, and `>` to `&gt;` - but that last `&gt;` is treated as an unescaped string even though it is really already an escaped `>` (which is why we get `&amp;gt;` at the end there).

It is definitely really weird, and I could probably go in and try fixing how the strings are escaped, but I feel like the possibility of introducing a weakness is not worth making the string prettier when we are not obligated to do so